app/components/Link/Link.tsx

```diff
@@ -0,0 +1,54 @@
+'use client';
+import { CustomLinkProps } from './Link.types';
+import { linkStyles } from './Link.styles';
+import Button from '@mui/material/Button';
+import { NextLinkComposed } from './NextLink';
+import Image from 'next/image';
+import { Typography } from '@mui/material';
+// import { Box } from '@mui/material';
+
+const CustomLink = ({ path, children, startIcon, endIcon, startSVG, endSVG }: CustomLinkProps) => {
+  const renderSVG = (svg: CustomLinkProps['startSVG']) => {
+    if (!svg) return null;
+
+    if (typeof svg === 'function') {
+      const SVGComponent = svg;
+      return <SVGComponent width={26} height={26} style={{ display: 'block' }} />;
```

Comment:
I think this is redundant. Use only `startIcon` and `endIcon`. Also you may move `SvgComponent` into shared components to make it reusable and simplify logic of the component. Does it make sense?